js/page-specific/search-results.js

```diff
@@ -0,0 +1,107 @@
+
+		if (current <= 3 && max <= 5) return buildArrayFromRange(1, max, current);
+
+		if (current <= 3 && max >= 5) return buildArrayFromRange(1, 5, current);
+
+		if (difference >= 2) return buildArrayFromRange(current - 2, current + 2, current);
+
+		if (max <= 5) return buildArrayFromRange(1, max, current);
+
+		return buildArrayFromRange(max - 4, max, current);
+	};
+
+	const searchSuccessHandler = (data) => {
+		const templateData = data;
+
+		if (templateData.SearchResults.length > 0) {
+			templateData.ResultsPerPageButtons = [
+				{ count: 10, isCurrent: data.ResultsPerPage === 10 },
+				{ count: 25, isCurrent: data.ResultsPerPage === 25 },
+				{ count: 50, isCurrent: data.ResultsPerPage === 50 },
+				{ count: 100, isCurrent: data.ResultsPerPage === 100 }
+			];
+
+			templateData.PageNumbers = getPageNumbers(templateData.CurrentPage, templateData.Pages);
+
+			getMaterialTypeData((materialTypes) => {
+				const recordsWithIcons = [];
+				_.map(templateData.SearchResults, (record) => {
+					const recordWithIcon = {};
+					recordWithIcon.icon = materialTypes[_.findIndex(materialTypes, {
+						id: parseInt(record.PrimaryTypeOfMaterial, 10)
+					})].icon;
+					recordsWithIcons.push(recordWithIcon);
+				});
+
+				templateData.SearchResults = recordsWithIcons;
+
+				const source = $('#search-results-template').html();
+				const template = Handlebars.compile(source);
+				const html = template(templateData);
+				$('.loading').hide();
+				$('#search-results').html(html);
+			});
+		} else {
+			$('.no-results').show();
+		}
+	};
+
+	const searchErrorHandler = (err) => {
+		console.log(err);
+	};
+
+	const resultCountButtonClicked = (event) => {
+		const $button = $(event.currentTarget);
+		const selectedResultsPerPage = $button.attr('data-results-per-page');
+
+		window.location = `${window.location.pathname}?q=${event.data.searchTerm}&page=${event.data.pageNumber}&resultsPerPage=${selectedResultsPerPage}`;
+	};
+
+	const pageButtonClicked = (event) => {
+		const $button = $(event.currentTarget);
+		const selectedPageNumber = $button.attr('data-page-number');
+
+		window.location = `${window.location.pathname}?q=${event.data.searchTerm}&page=${selectedPageNumber}&resultsPerPage=${event.data.resultsPerPage}`;
+	};
+
+	const init = () => {
+		const querystring = querystringer.getAsDictionary();
+
+		if (querystring.q) {
+			const searchTerm = querystring.q.trim();
+			const pageNumber = querystring.page ? querystring.page : 1;
+			const resultsPerPage = querystring.resultsperpage ? querystring.resultsperpage : 10;
+
+			$(document).on('click', '#pager .results-per-page button', { searchTerm, pageNumber, resultsPerPage }, resultCountButtonClicked);
+			$(document).on('click', '#pager .page-numbers button', { searchTerm, pageNumber, resultsPerPage }, pageButtonClicked);
+
+			$.get(`http://ba224964:1000/api/polaris/search/${searchTerm}/${pageNumber}/${resultsPerPage}`).then(searchSuccessHandler, searchErrorHandler);
```

Comment:
change url...probably should specify all url's near the top of each doc, or pass them in as an option in an init function or something like that.